include/seqan3/argument_parser/argument_parser.hpp

```diff
@@ -553,26 +573,33 @@ class argument_parser
             return;
         }
 
+        bool special_format_was_set{false};
+
         for(int i = 1, argv_len = argc; i < argv_len; ++i) // start at 1 to skip binary name
         {
             std::string arg{argv[i]};
 
+            if (std::ranges::find(subcommands, arg) != subcommands.end())
+            {
+                sub_parser = std::make_unique<argument_parser>(info.app_name + "-" + arg, argc - i, argv + i, false);
+                break;
```

Comment:
return here.

Reply:
If I return here I need to copy and paste the same if check as comes after the for loop into the current if scope. It is more dry like this.